src/main/java/me/xmrvizzy/skyblocker/skyblock/dungeon/DungeonChestProfit.java

```diff
@@ -0,0 +1,171 @@
+							}
+							
+							continue;
+						}
+						
+						//Determine if a kismet was used or not
+						if (name.contains("Reroll Chest")) {
+							usedKismet = !StringUtils.isBlank(searchLoreFor(stack, client, "You already rerolled a chest!"));
+							
+							continue;
+						}
+					}
+				}
+				
+				if (SkyblockerConfig.get().locations.dungeons.dungeonChestProfit.includeKismet && usedKismet) {
+					IntBooleanPair kismetPriceData = getItemPrice("KISMET_FEATHER");
+					
+					if (kismetPriceData.rightBoolean() == false) hasIncompleteData = true;
+					
+					profit -= kismetPriceData.leftInt();
+				}
+				
+				return ((MutableText) Text.literal(title.getString())).append(getProfitText(profit, hasIncompleteData));
+			}
+		} catch (Exception e) {
+			LOGGER.error("[Skyblocker Profit Calculator] Failed to calculate dungeon chest profit! ", e);
+		}
+		
+		return title;
+	}
+	
+	/**
+	 * 
+	 * @return An {@link IntBooleanPair} with the {@code left int} representing the item's price, and the {@code right boolean} indicating if the price
+	 * was based on complete data.
+	 */
+	private static IntBooleanPair getItemPrice(String id) {
+		JsonObject bazaarPrices = PriceInfoTooltip.getBazaarPrices();
+		JsonObject lbinPrices = PriceInfoTooltip.getLBINPrices();
+		
+		if (bazaarPrices == null || lbinPrices == null) return IntBooleanPair.of(0, false);
+		
+		if (bazaarPrices.has(id)) {
+			JsonObject item = bazaarPrices.get(id).getAsJsonObject();
+			boolean isPriceNull = item.get("sellPrice").isJsonNull();
+			
+			return IntBooleanPair.of(isPriceNull ? 0 : (int) item.get("sellPrice").getAsDouble(), !isPriceNull);
+		}
+		
+		if (lbinPrices.has(id)) {
+			return IntBooleanPair.of((int) lbinPrices.get(id).getAsDouble(), true);
+		}
+		
+		return IntBooleanPair.of(0, false);
+	}
+	
+	/**
+	 * Searches for a specific string of characters in the name and lore of an item
+	 */
+	private static String searchLoreFor(ItemStack stack, MinecraftClient client, String searchString) {
+		List<Text> lore = stack.getTooltip(client.player, TooltipContext.BASIC);
+		
+		for (int i = 0; i < lore.size(); i++) {
+			String line = lore.get(i).getString();
+			
+			if (line.contains(searchString)) {
+				return line;
+			}
+		}
+		
+		return null;
+	}
+	
+	private static boolean isDungeonChest(String name) {
+		return name.equals("Wood Chest") || name.equals("Gold Chest") || name.equals("Diamond Chest") || name.equals("Emerald Chest") || name.equals("Obsidian Chest") || name.equals("Bedrock Chest");
+	}
+	
+	private static Text getProfitText(int profit, boolean hasIncompleteData) {
+		return (profit == 0) ? Text.literal(" " + FORMATTER.format(profit)).setStyle(hasIncompleteData ? DARK_BLUE : GRAY) : (profit > 0) ? Text.literal(" +" + FORMATTER.format(profit)).setStyle(hasIncompleteData ? DARK_BLUE : DARK_GREEN) : Text.literal(" " + FORMATTER.format(profit)).setStyle(hasIncompleteData ? DARK_BLUE : DARK_RED);
+	}
```

Comment:
```suggestion
import net.minecraft.text.Style;
import net.minecraft.text.Text;
import net.minecraft.util.Formatting;

public class DungeonChestProfit {
    private static final Logger LOGGER = LoggerFactory.getLogger(DungeonChestProfit.class);
    private static final Pattern ESSENCE_PATTERN = Pattern.compile("(?<type>[A-Za-z]+) Essence x(?<amount>[0-9]+)");
    private static final DecimalFormat FORMATTER = new DecimalFormat("#,###");
    private static final Style GRAY = Style.EMPTY.withColor(Formatting.GRAY);
    private static final Style DARK_BLUE = Style.EMPTY.withColor(Formatting.DARK_BLUE);
    private static final Style DARK_GREEN = Style.EMPTY.withColor(Formatting.DARK_GREEN);
    private static final Style DARK_RED = Style.EMPTY.withColor(Formatting.DARK_RED);

    public static Text getChestProfit(GenericContainerScreenHandler handler, Text title, MinecraftClient client) {
        try {
            if (SkyblockerConfig.get().locations.dungeons.dungeonChestProfit.enableProfitCalculator && isDungeonChest(title.getString())) {
                int profit = 0;
                boolean hasIncompleteData = false, usedKismet = false;

                Slot openRewardChestSlot = handler.slots.get(31);
                Slot kismetSlot = handler.slots.get(50);

                if (openRewardChestSlot.getStack().isEmpty() || kismetSlot.getStack().isEmpty()) return title;

                for (Slot slot : handler.slots.subList(0, handler.getRows() * 9)) {
                    ItemStack stack = slot.getStack();
                    if (stack.isEmpty()) continue;

                    String name = stack.getName().getString();
                    String id = PriceInfoTooltip.getInternalNameFromNBT(stack, false);

                    if (id != null) {
                        IntBooleanPair priceData = getItemPrice(id);
                        if (!priceData.rightBoolean()) hasIncompleteData = true;
                        profit += priceData.leftInt();
                    } else if (name.contains("Essence")) {
                        Matcher matcher = ESSENCE_PATTERN.matcher(name);
                        if (matcher.matches()) {
                            String type = matcher.group("type");
                            int amount = Integer.parseInt(matcher.group("amount"));
                            IntBooleanPair priceData = getItemPrice(("ESSENCE_" + type).toUpperCase());
                            if (!priceData.rightBoolean()) hasIncompleteData = true;
                            profit += priceData.leftInt() * amount;
                        }
                    } else if (name.contains("Open Reward Chest")) {
                        String foundString = searchLoreFor(stack, client, "Coins");
                        if (!StringUtils.isBlank(foundString)) {
                            profit -= Integer.parseInt(foundString.replaceAll("[^0-9]", ""));
                        }
                    } else if (name.contains("Reroll Chest")) {
                        usedKismet = !StringUtils.isBlank(searchLoreFor(stack, client, "You already rerolled a chest!"));
                    }
                }

                if (SkyblockerConfig.get().locations.dungeons.dungeonChestProfit.includeKismet && usedKismet) {
                    IntBooleanPair kismetPriceData = getItemPrice("KISMET_FEATHER");
                    if (!kismetPriceData.rightBoolean()) hasIncompleteData = true;
                    profit -= kismetPriceData.leftInt();
                }

                return Text.literal(title.getString()).append(getProfitText(profit, hasIncompleteData));
            }
        } catch (Exception e) {
            LOGGER.error("[Skyblocker Profit Calculator] Failed to calculate dungeon chest profit! ", e);
        }

        return title;
    }

    /**
     * @return An {@link IntBooleanPair} with the {@code left int} representing the item's price, and the {@code right boolean} indicating if the price
     * was based on complete data.
     */
    private static IntBooleanPair getItemPrice(String id) {
        JsonObject bazaarPrices = PriceInfoTooltip.getBazaarPrices();
        JsonObject lbinPrices = PriceInfoTooltip.getLBINPrices();

        if (bazaarPrices == null || lbinPrices == null) return IntBooleanPair.of(0, false);

        if (bazaarPrices.has(id)) {
            JsonObject item = bazaarPrices.get(id).getAsJsonObject();
            boolean isPriceNull = item.get("sellPrice").isJsonNull();

            return IntBooleanPair.of(isPriceNull ? 0 : (int) item.get("sellPrice").getAsDouble(), !isPriceNull);
        }

        if (lbinPrices.has(id)) {
            return IntBooleanPair.of((int) lbinPrices.get(id).getAsDouble(), true);
        }

        return IntBooleanPair.of(0, false);
    }

    /**
     * Searches for a specific string of characters in the name and lore of an item
     */
    private static String searchLoreFor(ItemStack stack, MinecraftClient client, String searchString) {
        List<Text> lore = stack.getTooltip(client.player, TooltipContext.BASIC);

        for (Text text : lore) {
            String line = text.getString();

            if (line.contains(searchString)) {
                return line;
            }
        }

        return null;
    }

    private static boolean isDungeonChest(String name) {
        return name.equals("Wood Chest") || name.equals("Gold Chest") || name.equals("Diamond Chest") || name.equals("Emerald Chest") || name.equals("Obsidian Chest") || name.equals("Bedrock Chest");
    }

    private static Text getProfitText(int profit, boolean hasIncompleteData) {
        return (profit == 0) ? Text.literal(" " + FORMATTER.format(profit)).setStyle(hasIncompleteData ? DARK_BLUE : GRAY) : (profit > 0) ? Text.literal(" +" + FORMATTER.format(profit)).setStyle(hasIncompleteData ? DARK_BLUE : DARK_GREEN) : Text.literal(" " + FORMATTER.format(profit)).setStyle(hasIncompleteData ? DARK_BLUE : DARK_RED);
    }
```
You used tab instead of 4 spaces. And I have simplified a few lines of code.